### PR TITLE
Type definition for react-editable-label

### PR DESCRIPTION
**Pull request by markbiek**
This adds types for https://www.npmjs.com/package/react-editable-label (owned by me).

Eventually, I'd like to convert the package to Typescript and have the type definitions hosted internally.

But, for now, this is the quickest way to make the package functional for TS users.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by markbiek** [comment]
This is the error I seem to be getting:

```
Error: Package @types/react-editable-label is not marked as non-npm, but no implementation package was found on npm. If these types are not for an npm package, please add `"nonNpm": true` to the package.json. Otherwise, ensure the name of this package matches the name of the npm package.
```

Trying to figure out how to deal with ;)

**Comment by markbiek** [comment]
Not entirely sure why this is failing. As far as I can tell, the naming of the types package matches the npm package.

You can verify by running: `npm info react-editable-label`

**Comment by markbiek** [comment]
The error I posted above is from running `pnpm test react-editable-label`. But I'm getting that exact same error when I run `pnpm test <any other package>` so I don't think that's really the issue. Looking at the CI logs again.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-editable-label/react-editable-label-tests.tsx
1. types/react-editable-label/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2b486e9c67ba4cc039f7354408ff9201a3cc903d
</p>

**Comment by markbiek** [comment]
For anyone looking, the solution to my `Incorrect default export` error is in https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/68577

File: types/react-editable-label/tslint.json
Comment (jakebailey) [inline types/react-editable-label/tslint.json:1]: This file is no longer needed.

**Review by lindapaiste — APPROVED** [review]
It looks good to me, barring some really neurotic code style things in the tests:
- I would remove the `;` at the end of each line
- I would remove the brackets around your string inputs like `inputClass={"test-class"}`

**Comment by jakebailey** [comment]
This repo has a standard formatting which is applied after merge if not done in the PR. Semicolons are required. (I'm surprised that it isn't stricter about the JSX)

**Comment by lindapaiste** [comment]
> This repo has a standard formatting which is applied after merge if not done in the PR. Semicolons are required. (I'm surprised that it isn't stricter about the JSX)

I don’t know the particular formatting rules but semicolons really don’t make sense here, right?  In this situation the semicolon has no semantic meaning and does not end a statement.  It’s a character of text amid the other JSX HTML.  It’s `<Component/>{“;”}<Component/>{“;”}’.

**Comment by jakebailey** [comment]
Oh, you mean in the JSX; I didn't even notice those!

**Comment by markbiek** [comment]
The `;` were an artifact from an earlier attempt at the tests. I've removed those and the extra brackets.

**Comment by jakebailey** [comment]
I just noticed this in your PR description:

> This adds types for [npmjs.com/package/react-editable-label](https://www.npmjs.com/package/react-editable-label) (owned by me).

If you own the package, you can publish types yourself as a part of your package; you don't need to submit them to DT.

**Comment by markbiek** [comment]
> I just noticed this in your PR description:
> 
> > This adds types for [npmjs.com/package/react-editable-label](https://www.npmjs.com/package/react-editable-label) (owned by me).
> 
> If you own the package, you can publish types yourself as a part of your package; you don't need to submit them to DT.

I would like to do that eventually but this seemed like a faster way to get typing for the package in the meantime.

**Comment by jakebailey** [comment]
Theoretically it should only involve copying `index.d.ts` over to your repo (maybe with `package.json` `"types": "index.d.ts"` and publishing it, if you were thinking it was more complex than that.

**Comment by jakebailey** [comment]
I guess I'm not sure how I feel about merging a PR just to unmerge it soon after... If you're just looking for examples of packages which include a hand-written:

- https://unpkg.com/browse/chalk@4.1.2/
- https://unpkg.com/browse/source-map@0.7.4/
- https://unpkg.com/browse/klona@2.0.6/

File: types/react-editable-label/index.d.ts
Comment (jakebailey) [inline types/react-editable-label/index.d.ts:13]: Is this actually a named export? This looks like it'll be a default export (or `export =`?): https://github.com/markbiek/react-editable-label/blob/master/lib/EditableLabel.js#L93

But, not sure what the bundle looks like.

**Comment by markbiek** [comment]
> I guess I'm not sure how I feel about merging a PR just to unmerge it soon after... If you're just looking for examples of packages which include a hand-written:

Fair enough! Thanks for the examples. I'll take a look at those and see if I can update my package.

**Comment by jakebailey** [comment]
Of course, the benefit of DT is having people look at it, and the testing infra... (I did notice something not quite right in the types already.)

File: types/react-editable-label/index.d.ts
Comment (markbiek) [reply types/react-editable-label/index.d.ts:13]: Can you give me a little more detail? I'm not sure I completely understand what you're saying 😄.

Comment (jakebailey) [reply types/react-editable-label/index.d.ts:13]: Your readme says:

```
import EditableLabel from 'react-editable-label';
```

But this package's tests do:

```
import { EditableLabel } from 'react-editable-label';
```

Which are very different things. The declaration needs to have a `export default EditableLabel`, or more likely, an `export = EditableLabel`.

Comment (markbiek) [reply types/react-editable-label/index.d.ts:13]: Ah, I see what you're referring to. The only way I could get the DefinitelyTyped CI to pass was to change that export in the `index.d.ts`. When I move things to my package, I make sure the exports match.

Comment (jakebailey) [reply types/react-editable-label/index.d.ts:13]: Right, but we can't merge it here if it was still wrong, either.

**Pull request closed without merging**
